crates/alpen-ee/common/src/types/batch.rs

```diff
@@ -0,0 +1,110 @@
+use std::iter;
+
+use bitcoin::{Txid, Wtxid};
+use strata_acct_types::Hash;
+use strata_identifiers::L1BlockCommitment;
+
+use crate::ProofId;
+
+/// Unique, deterministic identifier for an Batch
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct BatchId {
+    prev_block: Hash,
+    last_block: Hash,
+}
+
+impl BatchId {
+    fn new(prev_block: Hash, last_block: Hash) -> Self {
+        Self {
+            prev_block,
+            last_block,
+        }
+    }
+}
+
+/// Batch-DA related data in an L1 block
+#[derive(Debug, Clone)]
+pub struct L1DaBlockRef {
+    /// L1 block holding DA txns.
+    pub block: L1BlockCommitment,
+    /// relevant transactions in this block.
+    pub txns: Vec<(Txid, Wtxid)>,
+    // inclusion merkle proof ?
+}
+
+/// Batch lifecycle states
+#[derive(Debug, Clone)]
+pub enum BatchStatus {
+    /// Newly created
+    Init,
+    /// DA txn(s) posted, waiting for inclusion in block.
+    DaPending { txns: Vec<(Txid, Wtxid)> },
+    /// DA txn(s) included in block(s).
+    DaComplete { da: Vec<L1DaBlockRef> },
+    /// Proving started, waiting for proof generation.
```

Comment:
there's a hybrid scenario where some txns are in blocks while other are not. how do you suggest we handle this?